Tests iOS/DeleteAnItemTests.swift

```diff
@@ -80,64 +76,45 @@ class DeleteAnItemTests: XCTestCase {
 
         itemCell.tap()
 
-        let expectRequest = expectation(description: "A request to the server")
-        server.routes.post("/graphql") { request, loop in
-            defer { expectRequest.fulfill() }
-            let apiRequest = ClientAPIRequest(request)
-            XCTAssertFalse(apiRequest.isEmpty)
-            XCTAssertTrue(apiRequest.isToDeleteAnItem)
-            XCTAssertTrue(apiRequest.contains("item-2"))
-
-            return Response.delete()
-        }
-
         app
             .readerView
+            .wait()
             .readerToolbar
+            .wait()
             .moreButton
             .wait()
             .tap()
 
         app.deleteButton.wait().tap()
         app.alert.yes.wait().tap()
-        wait(for: [expectRequest], timeout: 10)
+        wait(for: [deletionExpectation], timeout: 10)
 
         app.saves.wait()
         waitForDisappearance(of: itemCell)
     }
 
     func test_deletingAnItem_fromArchive_removesItFromList_andSyncsWithServer() {
-        app.tabBar.savesButton.wait().tap()
-        app.saves.wait().selectionSwitcher.archiveButton.wait().tap()
-        let cell = app.saves.itemView(matching: "Archived Item 1")
-
-        let receivedDeleteRequest = expectation(description: "receivedDeleteRequest")
+        let deletionExpectation = expectation(description: "A delete request to the server")
         server.routes.post("/graphql") { request, _ -> Response in
             let apiRequest = ClientAPIRequest(request)
-
-            if apiRequest.isForSlateLineup {
-                return .slateLineup()
-            } else if apiRequest.isForSavesContent {
-                return .saves()
-            } else if apiRequest.isForArchivedContent {
-                return .archivedContent()
-            } else if apiRequest.isToDeleteAnItem {
-                receivedDeleteRequest.fulfill()
-                XCTAssertFalse(apiRequest.isEmpty)
-                XCTAssertTrue(apiRequest.isToDeleteAnItem)
+            if apiRequest.isToDeleteAnItem {
+                defer { deletionExpectation.fulfill() }
                 XCTAssertTrue(apiRequest.contains("archived-item-1"))
-
                 return .delete()
-            } else {
-                fatalError("Unexpected request")
             }
+
+            return .fallbackResponses(apiRequest: apiRequest)
         }
+        app.launch()
+        app.tabBar.savesButton.wait().tap()
```

Comment:
would we ever need to also wait for `tabBar` (or any other ancestor in any other test) existence?

Reply:
I think waiting for the savesButton inherently waits for the tab bar but in my experience here... adding more waits does not hurt anything at all.